Assets/MRTK/Providers/WindowsMixedReality/Shared/InputHandlers/HPMotionControllerInputHandler.cs

```diff
@@ -0,0 +1,292 @@
+                MotionControllerHandedness controllerHandedness = controllerState.MotionController.Handedness;
+
+                Debug.Assert(controllerHandedness != MotionControllerHandedness.Unknown);
+                Debug.Assert(interactionMapping.AxisType == AxisType.Digital);
+
+                if (interactionMapping.InputType == DeviceInputType.TriggerTouch)
+                {
+                    var triggerData = controllerState.CurrentReading.GetPressedValue(ControllerInput.Trigger);
+                    interactionMapping.BoolData = !Mathf.Approximately(triggerData, 0.0f);
+                }
+                else if (interactionMapping.InputType == DeviceInputType.GripTouch)
+                {
+                    var gripData = controllerState.CurrentReading.GetPressedValue(ControllerInput.Grasp);
+                    interactionMapping.BoolData = !Mathf.Approximately(gripData, 0.0f);
+                }
+                else
+                {
+                    ControllerInput button;
+
+                    // Update the interaction data source
+                    // Interactions handled mirror the GenericXRSDKController to maintain parity. ThumbstickTouch and Touchpad are left out
+                    // due to having no ControllerInput equivalents
+                    switch (interactionMapping.InputType)
+                    {
+                        case DeviceInputType.Select:
+                        case DeviceInputType.TriggerNearTouch:
+                        case DeviceInputType.TriggerPress:
+                            button = ControllerInput.Trigger;
+                            break;
+                        case DeviceInputType.GripNearTouch:
+                        case DeviceInputType.GripPress:
+                            button = ControllerInput.Grasp;
+                            break;
+                        case DeviceInputType.ButtonPress:
+                        case DeviceInputType.PrimaryButtonPress:
+                            button = controllerHandedness == MotionControllerHandedness.Left ? ControllerInput.X_Button : ControllerInput.A_Button;
+                            break;
+                        case DeviceInputType.SecondaryButtonPress:
+                            button = controllerHandedness == MotionControllerHandedness.Left ? ControllerInput.Y_Button : ControllerInput.B_Button;
+                            break;
+                        case DeviceInputType.Menu:
+                            button = ControllerInput.Menu;
+                            break;
+                        case DeviceInputType.ThumbStickPress:
+                            button = ControllerInput.Thumbstick;
+                            break;
+                        default:
+                            return;
+                    }
+
+
+                    var buttonData = controllerState.CurrentReading.GetPressedValue(button);
+                    interactionMapping.BoolData = buttonData > 0.0f;
```

Comment:
I think we might want this to be 
```suggestion
                    interactionMapping.BoolData = Mathf.Approximately(buttonData, 1.0f);
```
instead of `> 0.0f`, since lines 107 and 112 establish that `!= 0.0f` is "touch", and "press" should be "all the way pressed"